### PR TITLE
ci: Add dockerfile and images for CI

### PR DESCRIPTION
**Pull request by nebril**

### CODE REVIEW

File: .github/workflows/images.yaml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+    steps:
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@2a4b53665e15ce7d7049afb11ff1f70ff1610609
+
+      - name: Login to quay.io for CI
+        uses: docker/login-action@f3364599c6aa293cdc2b8391b1b56d0c30e45c8a
+        with:
+          registry: quay.io
+          username: ${{ secrets.QUAY_CI_USERNAME }}
+          password: ${{ secrets.QUAY_CI_PASSWORD }}
```
Comment (aanm) [inline .github/workflows/images.yaml:37]: QUAY_CI_TOKEN

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ARG CILIUM_BUILDER_IMAGE=quay.io/cilium/cilium-builder:be4c0e477c4c8a61b3a41f6e8a0bab1d686ccd4e@sha256:0b69c4c6b680b6827c98167837a96647fa7d710912f2a001053b6c3bb0d4d9ae
+
+FROM ${CILIUM_BUILDER_IMAGE} as builder
+WORKDIR /go/src/github.com/cilium/cilium-cli
+COPY . .
+RUN make
+
+FROM bitnami/kubectl:1.16.15-debian-10-r220@sha256:2c35939baa0105d842f6ee144d81b083fba7b2d54f12d10025c243fa75a72a9b as kubectl
+
+FROM busybox:stable-glibc@sha256:78089367dc288a2d7b0c40802497e8ef99c5b1536344388dd5c628f1a40abb20
```
Comment (aanm) [inline Dockerfile:15]: `docker.io/library/busybox`

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+# SPDX-License-Identifier: Apache-2.0
+
+ARG CILIUM_BUILDER_IMAGE=quay.io/cilium/cilium-builder:be4c0e477c4c8a61b3a41f6e8a0bab1d686ccd4e@sha256:0b69c4c6b680b6827c98167837a96647fa7d710912f2a001053b6c3bb0d4d9ae
+
+FROM ${CILIUM_BUILDER_IMAGE} as builder
+WORKDIR /go/src/github.com/cilium/cilium-cli
+COPY . .
+RUN make
+
+FROM bitnami/kubectl:1.16.15-debian-10-r220@sha256:2c35939baa0105d842f6ee144d81b083fba7b2d54f12d10025c243fa75a72a9b as kubectl
```
Comment (aanm) [inline Dockerfile:13]: regarding `kubectl`, we can download it directly from `https://dl.k8s.io/release/${k8s_version}/bin/linux/amd64/kubectl` where `k8s_version` is something that we can provide

**Review by aanm — CHANGES_REQUESTED** [review]

**Comment by nebril** [comment]
The image has been built, once the review is green I will convert it to use `pull_request_target`

**Review by aanm — APPROVED** [review]

**Pull request merged**